src/components/userProfile/userProfile.js

```diff
@@ -55,61 +88,150 @@ class UserProfilePage extends React.Component {
                         <div className="infoName">
                             <p> Full name </p>
                         </div>
-                        <div className="infoInput">
-                            <Input value={userData.name + ' ' + userData.surname}/>
-                        </div>
-                    </div>
 
-                    <div className="info">
-                        <div className="infoName">
-                            <p> Name </p>
-                        </div>
-                        <div className="infoInput">
-                            <Input value={userData.name}/>
+                        <div>
+                            <p>{userData.name + ' ' + userData.surname}</p>
                         </div>
                     </div>
 
-                    <div className="info">
-                        <div className="infoName">
-                            <p> Surname </p>
-                        </div>
-                        <div className="infoInput">
-                            <Input value={userData.surname}/>
+                    <Form
+                        onFinish={this.onFinish}
+                        onFinishFailed={this.onFinishFailed}
+                    >
+                        <div className="info">
+                            <div className="infoName">
+                                <p> Name </p>
+                            </div>
+
+                            <div className="inputBlock">
+                                <Form.Item className="formItem" name="name"
+                                           initialValue={userData.name}
+                                           rules={[
+                                               {
+                                                   type: "string",
+                                                   pattern: new RegExp("^[A-Z][a-z]+$"),
```

Comment:
In my opinion, this pattern should be made in a separate file.

Reply:
So far we haven't found a way to output these patterns to another file correctly.  You can write your idea in response

Reply:
Ragex must be incapsulated in object